### PR TITLE
RBAC: Add origin column to seed_assignment

### PR DESCRIPTION
**Pull request by gamab**
**What is this feature?**

We've had some issues with failed plugin provisioning wrongfully triggering the removal of plugin roles and their permission assignments.

This PR:
* adds a new column to the seed table to store the origin of the seeded permissions
* adds a migration to OnCall permissions (which are the only plugin permissions so far)

**Why do we need this feature?**

[Add a description of the problem the feature is trying to solve.]

**Who is this feature for?**

[Add information on what kind of user the feature is for.]

**Which issue(s) does this PR fix?**:

Fixes #

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: pkg/services/sqlstore/migrations/accesscontrol/seed_assignment.go [hunk 1]
```diff
@@ -119,3 +125,21 @@ func (m *seedAssignmentPrimaryKeyMigrator) Exec(sess *xorm.Session, mig *migrato
...
+type seedAssignmentOnCallMigrator struct {
+	migrator.MigrationBase
+}
+
+func (m *seedAssignmentOnCallMigrator) SQL(dialect migrator.Dialect) string {
+	return CodeMigrationSQL
+}
+
+func (m *seedAssignmentOnCallMigrator) Exec(sess *xorm.Session, mig *migrator.Migrator) error {
+	_, err := sess.Exec(
```
[the remark this answers is not in the document]
Comment (gamab) [reply pkg/services/sqlstore/migrations/accesscontrol/seed_assignment.go:138]: Also it does not work in the upgrade>downgrade>upgrade case, but it does not feel right to run it on every start up.

File: pkg/services/sqlstore/migrations/accesscontrol/seed_assignment.go [hunk 2]
```diff
@@ -119,3 +125,21 @@ func (m *seedAssignmentPrimaryKeyMigrator) Exec(sess *xorm.Session, mig *migrato
...
+		`UPDATE seed_assignment SET origin = ? WHERE action LIKE ? OR scope = ?`,
+		"grafana-oncall-app",
```
Comment (kalleep) [inline pkg/services/sqlstore/migrations/accesscontrol/seed_assignment.go:140]: This would need to use a wildcard right?

**Review by Jguer — APPROVED** [review]

File: pkg/services/accesscontrol/pluginutils/utils.go [hunk 3]
```diff
@@ -60,6 +60,22 @@ func ToRegistrations(pluginID, pluginName string, regs []plugins.RoleRegistratio
...
+		return ""
+	}
+
+	pluginID := strings.Builder{}
+	for _, c := range roleName[len(ac.PluginRolePrefix):] {
+		if c == ':' {
+			break
+		}
+		pluginID.WriteRune(c)
+	}
```
Comment (IevaVasiljeva) [inline pkg/services/accesscontrol/pluginutils/utils.go:75]: nit: could just split on `:` and take the second element. Would be slightly more readable to me, but same result.

**Review by IevaVasiljeva — APPROVED** [review]
LGTM. Regarding upgrade -> downgrade issue with the migration, I would hope that people don't assign plugin permissions too often.

**Pull request merged**